eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -2034,7 +2034,7 @@ private function deleteTranslationFromContentObject($contentId, $languageId)
             ->where('id = :contentId')
             ->andWhere(
             // make sure removed translation is not the last one (incl. alwaysAvailable)
-                $query->expr()->orX(
+                $query->expr()->andX(
                     'language_mask & ~ ' . $languageId . ' <> 0',
                     'language_mask & ~ ' . $languageId . ' <> 1'
                 )
```

Comment:
Well... this is embarrassing, but the where clause didn't contain proper condition. This is just a precaution against some race condition that shouldn't really occur (that's why I've missed my mistake initially).

A little bit out of scope, but so small I've decided to include it as well.